core/src/main/scala/ai/h2o/sparkling/backend/shared/SharedBackendConf.scala

```diff
@@ -471,4 +493,23 @@ object SharedBackendConf {
 
   /** Whether certificates should be verified before using in H2O or not. */
   val PROP_VERIFY_SSL_CERTIFICATES: (String, Boolean) = ("spark.ext.h2o.verify_ssl_certificates", true)
+
+  /**
+   * If enabled, H2O instances will create JDBC connections to Hive and H2O Python & R API will be able to read data
+   * from hive. Don't forget to put a jar with Hive driver on spark classpath if the internal backend is used.
+   */
+  val PROP_HIVE_SUPPORT_ENABLED: (String, Boolean) = ("spark.ext.h2o.hive.enabled", false)
+
+  /** The full address of HiveServer2, for example hostname:10000 */
+  val PROP_HIVE_HOST: (String, None.type) = ("spark.ext.h2o.hive.host", None)
+
+  /** Hiveserver2 Kerberos principal, for example hive/hostname@DOMAIN.COM */
+  val PROP_HIVE_PRINCIPAL: (String, None.type) = ("spark.ext.h2o.hive.principal", None)
+
+  /** Can be used to further customize the way the driver constructs the Hive JDBC URL */
+  val PROP_HIVE_JDBC_URL_PATTERN: (String, String) =
+    ("spark.ext.h2o.hive.jdbc_url_pattern", "jdbc:hive2://{{host}}/;{{auth}}")
```

Comment:
do we need to duplicate the default value here?

Reply:
Where is the second definition? In H2O-3 codebase? Is this property optional https://github.com/h2oai/sparkling-water/pull/2000/files/353b70dd3f6362c1310ac8733dfff13c0823309a#diff-e3983f89666b52c7f5935a5ea1a6c29cR196 if we call `DelegationTokenRefresher.setup`?

Reply:
yes, this property is optional. BUT: either urlPattern OR hiveHost must be set

Reply:
default is defined
https://github.com/h2oai/h2o-3/blob/5161c70bff9e717d6c2f1b310598aa7de6437dd3/h2o-hive/src/main/java/water/hive/HiveTokenGenerator.java#L32
and here
https://github.com/h2oai/h2o-3/blob/5161c70bff9e717d6c2f1b310598aa7de6437dd3/h2o-hive/src/main/java/water/hive/HiveTokenGenerator.java#L45

Reply:
@honzasterba I've made the changes and the default values is not defined in SW anymore. Could you double check that [these conditions](https://github.com/h2oai/sparkling-water/pull/2000/files#diff-8007bc0a47fbb7c1aa706403041431f4R114-R129) are correct?

Reply:
lgtm